src/pages/api/stripe/webhook.ts

```diff
@@ -17,8 +33,65 @@ const stripeWebhookHandler = async (
     try {
       event = stripe.webhooks.constructEvent(buf, sig as string, webhookSecret)
 
-      if (event.type === 'checkout.session.completed') {
-        console.log('stripe webhook handler', {event})
+      // Also handle:
+      // - 'customer.subscription.updated'
+      // - 'customer.subscription.deleted'
+      if (event.type === 'customer.subscription.created') {
+        // this types it as a Stripe.Subscription instead of any
+        const stripeSubscription = await stripe.subscriptions.retrieve(
+          event.data.object.id,
+        )
+        const stripeCustomer = await stripe.customers.retrieve(
+          event.data.object.customer,
+        )
+
+        if (!('email' in stripeCustomer)) {
+          // solves a typescript error 🤡
+          throw new Error('No email found for customer')
+        }
+
+        const stripeEmail = stripeCustomer.email
+
+        // we want to get the `co_************` id from CustomerIO which we can
+        // cross-reference with MixPanel to uniquely identify the user.
+        // This correlates to the contact_id of the user in egghead.io
+        let cioCustomer = await cioAxios
```

Comment:
I'm on the fence here and think of the egghead db as canonical but we don't currently fetch `Contact` by email anywhere. The `Contact#id` would be this same value.